app/models/users/user.rb

```diff
@@ -5,43 +5,69 @@ class User < ActiveRecord::Base
   belongs_to :grove
 
   def self.from_omniauth(access_token)
-    data = access_token.info
-    if within_roots?(access_token)
-      if Teacher.is_teacher?(data)
-        user = Teacher.where(email: data["email"]).first
-        user = check_or_create_token(user, access_token)
-      elsif Student.is_student?(data)
-        user = Student.where(email: data["email"]).first
-        user = check_or_create_token(user, access_token)
-      end
-      user
+    package = OmniauthPackage.new(access_token)
+    user = User.where(email: package.email).first
```

Comment:
would this line blow up if someone tried to sign in with a google account that's not already in the system? @jedschneider